### DISCUSSION TITLE
Prevent drop things from outside of grapesjs

### DISCUSSION THREAD

**Discussion by ronaldohoch** [GrapesJS/grapesjs#5242] [Q&A] [answered]
Hello!

I can drop things from outside of grapesjs, things like text and images. From files or other websites.

Is there a way to prevent this?

Thank you.

**Comment by artf** [comment, marked as answer]
Right now you can rely on something like `canvas:dragdata` and `dragContent` to understand if the content to drop is an external one
```js
editor.on('canvas:dragdata', (dataTransfer, result) => {
  const isExternal = !editor.getModel().get('dragContent');
  if (isExternal) {
    // with a falsy `content`, nothing will be created
    result.content = null;
   }
});
```
The only problem is the editor would still allow to choice where to drop the content.

I'll try to see if it's possible to introduce a new config option that prevents, completely, external elements (eg. `config.canvas.allowExternalDrop`)

**Reply by ronaldohoch** [reply]
Thank you a lot!
